HerPublicWebsite/Controllers/QuestionnaireController.cs

```diff
@@ -453,17 +453,57 @@ public async Task<IActionResult> Eligible_Post(EligibleViewModel viewModel)
 
         return RedirectToNextStep(nextStep, viewModel.EntryPoint);
     }
-    
+
     [HttpGet("confirmation")]
     public IActionResult Confirmation_Get()
     {
         return RedirectToAction(nameof(StaticPagesController.Index), "StaticPages");
     }
 
     [HttpGet("ineligible")]
-    public IActionResult Ineligible_Get()
+    public IActionResult Ineligible_Get(bool submitted = false)
     {
-        return RedirectToAction(nameof(StaticPagesController.Index), "StaticPages");
+        var questionnaire = questionnaireService.GetQuestionnaire();
+        var viewModel = new IneligibleViewModel
+        {
+            FoundEpcIsTooHigh = questionnaire.FoundEpcBandIsTooHigh && (questionnaire.EpcDetailsAreCorrect ?? true),
+            IncomeBand = questionnaire.IncomeBand,
+            LocalAuthorityName = questionnaire.LocalAuthorityName,
+            LocalAuthorityWebsite = questionnaire.LocalAuthorityWebsite,
+            Submitted = submitted,
+            BackLink = GetBackUrl(QuestionFlowStep.Ineligible, questionnaire)
+        };
+
+        return View("Ineligible", viewModel);
+    }
+
+    [HttpPost("ineligible")]
+    public async Task<IActionResult> Ineligible_Post(IneligibleViewModel viewModel)
+    {
+        if (!ModelState.IsValid)
+        {
+            return Ineligible_Get();
+        }
+
+        if (viewModel.CanContactByEmailAboutFutureSchemes is YesOrNo.Yes)
+        {
+            await questionnaireService.AddFutureNotificationContactDetailsAsync(viewModel.EmailAddress);
+        }
+
+        return Ineligible_Get(true);

```

Comment:
After a successful post you need to redirect so that if the user refreshes the page it doesn't resubmit the form.

You can add the query parameter like you did with the address page, or look at `Confirmation_Post` from 554